testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/TwoMixedHives.java

```diff
@@ -83,7 +87,7 @@ public void extendEnvironment(Environment.Builder builder)
     @SuppressWarnings("resource")
     private DockerContainer createHadoopMaster2()
     {
-        return createHadoopContainer(dockerFiles, hadoopBaseImage + ":" + hadoopImagesVersion, HADOOP + "-2")
+        return createHadoopContainer(dockerFiles, portBinder, hadoopBaseImage + ":" + hadoopImagesVersion, HADOOP + "-2")
```

Comment:
I don't think it will work to expose ports from two different containers on the same host ports.

Reply:
It won't work without this change as we are adding health check for open port 10000 on hadoop containers and this won't pass new validation in the testcontainers (as port is not exposed). Please note that on CI we are not exposing ports at all, so containers can have overlapping ports.

Reply:
But you are not able to use `--debug` switch to run this environment. Maybe we need to expose them on random port when `--debug` is not used.

Reply:
Also I am concerned that we need to expose ports to make health checks to run. Can we use containers internal ports for that?

Reply:
I've reworked that part: PTAL